src/transformers/modeling_gpt2.py

```diff
@@ -145,7 +145,7 @@ def _attn(self, q, k, v, attention_mask=None, head_mask=None):
             w = w / (v.size(-1) ** 0.5)
         nd, ns = w.size(-2), w.size(-1)
         mask = self.bias[:, :, ns - nd : ns, :ns]
-        w = torch.where(mask, w, self.masked_bias)
+        w = torch.where(mask, w, self.masked_bias.to(w.dtype))
```

Comment:
```suggestion
        masked_bias = self.masked_bias.to(w.dtype)
        assert masked_bias.item() != -float("inf"), "Make sure `self.masked_bias` is not `-inf` in fp16 mode"
        w = torch.where(mask, w, self.masked_bias.to(w.dtype))
```

Reply:
One has to be careful with `fp16` and large numbers: 
```python
torch.tensor(-1e4).to(torch.float16).item() == -float('inf') #False so all good, but
torch.tensor(-1e5).to(torch.float16).item() == -float('inf') #4058 
```

Reply:
It's set to 1e-4 in __init__, so we're OK.

Reply:
Yeah, think it's still saver to do this in case (for whatever reason) someone changes the mask value to < 1e-4

Reply:
Are you adding this additional assert because you think the exact `-1e4` value is important?

If it's for this reason, then you can remove the test because the original GPT2 (from which we load the weights) had the real value `-1e10` anyway (see [here](https://github.com/openai/gpt-2/blob/master/src/model.py#L88)) so `1e4` had not specific reason to be this value apart from working well with fp16 (before we switched to using `torch.where` which is the real reason this bug appeared).

Reply:
I was thinking that if people might use the model and change the mask for whatever reason, they are not surprised that `-inf` comes out in fp16.